common-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala

```diff
@@ -15,33 +15,44 @@ object MessageListener {
 
   type MsgParser[T <: BusMessage] = T => Future[_]
 
-  def buildSource[T <: BusMessage](fromSource: Source[CommittableMessage[Array[Byte], T], NotUsed],
+  /**
+    * This trait is used to enable NATS to pretend it can do committable messages
+    */
+  trait CommittableMessageLike[T <: BusMessage] {
+    def getMsg(): T
+    def commitRecord(): Future[Done]
+  }
+
+  class NatsMsg[T <: BusMessage](msg: T) extends CommittableMessageLike[T] {
+    override def getMsg(): T = msg
+
+    override def commitRecord(): Future[Done] = Future.successful(Done)
+  }
+
+  class KafkaMsg[T <: BusMessage](msg: CommittableMessage[_, T]) extends CommittableMessageLike[T] {
+    override def getMsg(): T = msg.record.value()
+
+    override def commitRecord(): Future[Done] = msg.committableOffset.commitScaladsl()
+  }
+
+  def buildSource[T <: BusMessage](fromSource: Source[CommittableMessageLike[T], NotUsed],
                                    op: MsgParser[T])
                                   (implicit system: ActorSystem, ml: MessageLike[T]): Source[T, NotUsed] = {
     implicit val ec = system.dispatcher
 
     fromSource
-      .mapAsync(3) { commitableMsg =>
-        val msg = commitableMsg.record.value()
-        op(msg).map(_ => commitableMsg)
+      .mapAsync(3) { committableMsg =>
+        val msg = committableMsg.getMsg()
+        op(msg).map(_ => committableMsg)
       }
-      .mapAsync(1) { commitableMsg =>
-        commitableMsg.committableOffset.commitScaladsl().map(_ => commitableMsg.record.value())
+      .mapAsync(1) { committableMsg =>
+        committableMsg.commitRecord().map(_ => committableMsg.getMsg())
       }
   }
 
-  private def kafkaSource[T <: BusMessage](config: Config)
-                                          (implicit system: ActorSystem, ml: MessageLike[T])
-  : Source[CommittableMessage[Array[Byte], T], NotUsed] =
-    KafkaClient.commitableSource[T](config)(ml, system) match {
-      case Xor.Right(s) => s
-      case Xor.Left(err) => throw err
-    }
-
   def props[T <: BusMessage](config: Config, op:MsgParser[T])
                             (implicit system: ActorSystem, ml: MessageLike[T]): Props = {
-    val source = buildSource(kafkaSource(config), op)
+    val source = buildSource(MessageBus.subscribeCommittable(config), op)
```

Comment:
That's annoying. I think this is just an unfortunate consequence of silently kinda dropping the need to set `messaging.mode` when this in fact is still needed.